`test_sid_resolution` failed : 
```
2026-06-10T10:33:50.147 INFO:journalctl@ceph.mon.a.trial049.stdout:Jun 10 10:33:49 trial049 bash[10532]: audit 2026-06-10T10:33:49.724073+0000 mon.a (mon.0) 549 : audit [DBG] from='mgr.14287 10.20.193.49:0/208334764' entity='mgr.x' cmd={"prefix": "osd tree", "states": ["destroyed"], "format": "json"} : dispatch
2026-06-10T10:33:50.828 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_sid_resolution.py::test_sid_resolution FAILED                 [ 50%]
2026-06-10T10:33:50.857 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_simple_smb.py::test_arbitrary_listdir[share1] PASSED          [ 58%]
2026-06-10T10:33:50.883 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_simple_smb.py::test_arbitrary_listdir[share2] PASSED          [ 66%]
2026-06-10T10:33:50.917 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_simple_smb.py::test_create_dir[share1] PASSED                 [ 75%]
2026-06-10T10:33:50.950 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_simple_smb.py::test_create_dir[share2] PASSED                 [ 83%]
2026-06-10T10:33:50.983 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_simple_smb.py::test_create_file[share1] PASSED                [ 91%]
2026-06-10T10:33:51.036 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_simple_smb.py::test_create_file[share2] PASSED                [100%]
2026-06-10T10:33:51.040 INFO:tasks.workunit.client.0.trial049.stdout:
2026-06-10T10:33:51.043 INFO:tasks.workunit.client.0.trial049.stdout:=================================== FAILURES ===================================
2026-06-10T10:33:51.046 INFO:tasks.workunit.client.0.trial049.stdout:_____________________________ test_sid_resolution ______________________________
2026-06-10T10:33:51.049 INFO:tasks.workunit.client.0.trial049.stdout:
2026-06-10T10:33:51.052 INFO:tasks.workunit.client.0.trial049.stdout:smb_cfg = <smbutil.SMBTestConf object at 0x7eff81d65d80>
2026-06-10T10:33:51.055 INFO:tasks.workunit.client.0.trial049.stdout:
2026-06-10T10:33:51.058 INFO:tasks.workunit.client.0.trial049.stdout:    @pytest.mark.domain
2026-06-10T10:33:51.061 INFO:tasks.workunit.client.0.trial049.stdout:    def test_sid_resolution(smb_cfg):
2026-06-10T10:33:51.063 INFO:tasks.workunit.client.0.trial049.stdout:        """Verify that rpcclient lookupsids resolves domain user SIDs correctly
2026-06-10T10:33:51.066 INFO:tasks.workunit.client.0.trial049.stdout:        inside the smbd container, preventing regressions on /run bind mount
2026-06-10T10:33:51.068 INFO:tasks.workunit.client.0.trial049.stdout:        permissions that break smbd to winbindd communication (tracker#77120).
2026-06-10T10:33:51.070 INFO:tasks.workunit.client.0.trial049.stdout:        """
2026-06-10T10:33:51.073 INFO:tasks.workunit.client.0.trial049.stdout:        cluster_id = smbutil.get_shares(smb_cfg)[0]['cluster_id']
2026-06-10T10:33:51.075 INFO:tasks.workunit.client.0.trial049.stdout:        username = smb_cfg.username
2026-06-10T10:33:51.077 INFO:tasks.workunit.client.0.trial049.stdout:        password = smb_cfg.password
2026-06-10T10:33:51.079 INFO:tasks.workunit.client.0.trial049.stdout:
2026-06-10T10:33:51.080 INFO:tasks.workunit.client.0.trial049.stdout:>       result = cephutil.cephadm_enter_cmd(
2026-06-10T10:33:51.082 INFO:tasks.workunit.client.0.trial049.stdout:            smb_cfg,
2026-06-10T10:33:51.084 INFO:tasks.workunit.client.0.trial049.stdout:            cluster_id,
2026-06-10T10:33:51.086 INFO:tasks.workunit.client.0.trial049.stdout:            ['wbinfo', '-n', username],
2026-06-10T10:33:51.088 INFO:tasks.workunit.client.0.trial049.stdout:            capture_output=True,
2026-06-10T10:33:51.092 INFO:tasks.workunit.client.0.trial049.stdout:            check=True,
2026-06-10T10:33:51.094 INFO:tasks.workunit.client.0.trial049.stdout:        )
2026-06-10T10:33:51.097 INFO:tasks.workunit.client.0.trial049.stdout:
2026-06-10T10:33:51.100 INFO:tasks.workunit.client.0.trial049.stdout:tests/test_sid_resolution.py:17:
2026-06-10T10:33:51.103 INFO:tasks.workunit.client.0.trial049.stdout:_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2026-06-10T10:33:51.105 INFO:tasks.workunit.client.0.trial049.stdout:tests/cephutil.py:90: in cephadm_enter_cmd
2026-06-10T10:33:51.108 INFO:tasks.workunit.client.0.trial049.stdout:    return subprocess.run(cmd, **kwargs)
2026-06-10T10:33:51.110 INFO:tasks.workunit.client.0.trial049.stdout:_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2026-06-10T10:33:51.112 INFO:tasks.workunit.client.0.trial049.stdout:
```

Logs: https://qa-proxy.ceph.com/teuthology/avan-2026-06-10_09:59:08-orch:cephadm:smb-wip-athakkar-testing-2026-06-10-1328-distro-default-trial/282616/teuthology.log

--- a/qa/suites/orch/cephadm/smb/tasks/deploy_smb_mgr_domain.yaml
+++ b/qa/suites/orch/cephadm/smb/tasks/deploy_smb_mgr_domain.yaml
@@ -62,7 +62,7 @@ tasks:
     timeout: 1h
     clients:
       client.0:
-        - [default, hosts_access]
+        - [default, hosts_access, domain]
 
 - cephadm.shell:
     host.a:

--- a/qa/workunits/smb/tests/cephutil.py
+++ b/qa/workunits/smb/tests/cephutil.py
@@ -1,5 +1,6 @@
 import enum
 import json
+import shlex
 import subprocess
 
 
@@ -66,3 +67,27 @@ def cephadm_shell_cmd(
     elif load is LoadJSON.ERROR:
         return JSONResult(proc.returncode, None, proc.stderr.decode())
     return proc
+
+
+def cephadm_enter_cmd(smb_cfg, cluster_id, args, **kwargs):
+    """Run a command inside the primary smbd container for the given
+    cluster_id on the cluster's admin node (derived via smb_cfg).
+    All kwargs are treated as arguments to subprocess.run.
+    """
+    remote_cmd = [
+        'sudo',
+        f'/home/{smb_cfg.ssh_user}/cephtest/cephadm',
+        'enter',
+        '-i',
+        f'smb.{cluster_id}',
+    ] + list(args)
+    cmd = [
+        'ssh',
+        '-oBatchMode=yes',
+        '-oUserKnownHostsFile=/dev/null',
+        '-oStrictHostKeyChecking=no',
+        '-q',
+        f'{smb_cfg.ssh_user}@{smb_cfg.ssh_admin_host}',
+        shlex.join(remote_cmd),
+    ]
+    return subprocess.run(cmd, **kwargs)

--- a/qa/workunits/smb/tests/pytest.ini
+++ b/qa/workunits/smb/tests/pytest.ini
@@ -6,3 +6,4 @@ markers =
     hosts_access: Host access tests
     rate_limiting: Rate limit tests
     ceph_smb_ctl_local: Local/container test of ceph-smb-ctl tool
+    domain: Domain integration tests

--- a/qa/workunits/smb/tests/test_sid_resolution.py
+++ b/qa/workunits/smb/tests/test_sid_resolution.py
@@ -1,0 +1,46 @@
+import pytest
+
+import cephutil
+import smbutil
+
+
+@pytest.mark.domain
+def test_sid_resolution(smb_cfg):
+    """Verify that rpcclient lookupsids resolves domain user SIDs correctly
+    inside the smbd container, preventing regressions on /run bind mount
+    permissions that break smbd to winbindd communication (tracker#77120).
+    """
+    cluster_id = smbutil.get_shares(smb_cfg)[0]['cluster_id']
+    username = smb_cfg.username
+    password = smb_cfg.password
+
+    result = cephutil.cephadm_enter_cmd(
+        smb_cfg,
+        cluster_id,
+        ['wbinfo', '-n', username],
+        capture_output=True,
+        check=True,
+    )
+    user_sid = result.stdout.decode().split()[0]
+    assert user_sid.startswith('S-'), f'unexpected SID format: {user_sid}'
+
+    auth = f'{username}%{password}'
+    result = cephutil.cephadm_enter_cmd(
+        smb_cfg,
+        cluster_id,
+        [
+            'rpcclient',
+            'localhost',
+            '-U',
+            auth,
+            '-c',
+            f'lookupsids {user_sid}',
+        ],
+        capture_output=True,
+        check=True,
+    )
+    output = result.stdout.decode()
+    short_name = username.split('\\')[-1]
+    assert short_name in output, (
+        f'SID resolution failed: {short_name!r} not found in: {output}'
+    )

